src/app/client/src/app/app.component.ts

```diff
@@ -122,7 +123,42 @@ export class AppComponent implements OnInit, OnDestroy {
           _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideHeaderNFooter');
       });
   }
+  public makeUTMSession(params) {
+    for (const item in params) {
```

Comment:
can you write this logic in functional and concise way. Using ENUM to map utm params.
```
enum UTM_PARAMS {
   channel = 'Source';
   utm_campaign = 'Source';
   utm_medium = 'UtmMedium'
}
_.toPairs(param).filter(([key, value]) => UTM_ENUM[key]).map(([key, value]) => ({id: key, value: UTM_ENUM[value]}))
```

Reply:
Done implementing as you suggested.